### PR TITLE
vlc: bump to 3.0.7

### PR DESCRIPTION
**Pull request by alarcher**
Update from 2.x requires running /usr/lib/amd64/vlc/vlc-cache-gen /usr/lib/amd64/vlc/plugins after install to regenerate the plugins cache.

Works fine but OpenGL output needs love.

### CODE REVIEW

**Comment by alarcher** [comment]
Required after upgrade:

helios> sudo /usr/lib/amd64/vlc/vlc-cache-gen /usr/lib/amd64/vlc/plugins

**Comment by pyhalov** [comment]
Should we wrap it in smf service?

**Comment by alarcher** [comment]
That's probably the best option yes. I have never done that though so if you have an example (desktop-cache?) that would be great.

**Comment by pyhalov** [comment]
Yes, desktop-cache is a good example.

**Comment by pyhalov** [comment]
@alarcher Do you have time to resurrect this PR?

**Comment by pyhalov** [comment]
Does it require updated ffmeg?

**Comment by pyhalov** [comment]
vlc.p5m is not included....

**Comment by alarcher** [comment]
@pyhalov Does not build anymore right now

**Comment by pyhalov** [comment]
builds here, publish fails (qt5 runpath). Likely, we have to patch Qt5 (--libs doesn't include -R).

**Comment by pyhalov** [comment]
Look at https://forum.videolan.org/viewtopic.php?t=134770 .
When VLC uses ffmpeg, it uses multi-threading, but disables hardware acceleration (vdpau). 
You have to change it manually when using NVidia adapter.

Otherwise looks good.
I'm going to merge ffmpeg update (https://github.com/OpenIndiana/oi-userland/pull/4034), 
followed by https://github.com/OpenIndiana/oi-userland/pull/4428 , rebuild of audacity and this PR tomorrow.

**Comment by pyhalov** [comment]
vlc-cache-gen service is incorrect, it should be more like desktop-cache services.

**Comment by pyhalov** [comment]
Cache file is /usr/lib/amd64/vlc/plugins/plugins.dat

**Comment by pyhalov** [comment]
@alarcher : Please, test. I consider this ready now.

File: components/encumbered/vlc/Makefile [hunk 1]
```diff
@@ -37,19 +36,33 @@ include $(WS_MAKE_RULES)/prep.mk
 include $(WS_MAKE_RULES)/configure.mk
 include $(WS_MAKE_RULES)/ips.mk
 
+PATH=$(PATH.gnu)
```
Comment (Mno-hime) [inline components/encumbered/vlc/Makefile:39]: It's there again on line 62.

File: components/encumbered/vlc/patches/04-nvidia-hw.patch [hunk 2]
```diff
@@ -0,0 +1,152 @@
+https://forum.videolan.org/viewtopic.php?t=134770
+Avcodec doesn't enable HW acceleration by default in multi-threaded setup.
+And threading is selected automatically based on CPU counts.
+However, when HW acceleration is available, it's much more efficient.
+So we enable it for NVidia adapters.
+On platforms without HW acceleration (like ATI or modern Intel video adapters, for wich we lack drivers)
```
Comment (Mno-hime) [inline components/encumbered/vlc/patches/04-nvidia-hw.patch]: s/wich/which/

**Review by Mno-hime — COMMENTED** [review]
Will test this weekend.

**Comment by Mno-hime** [comment]
I don't see H.264 accelerated on Intel Ivy Bridge GPU (Lenovo X230) with updated FFMPEG (https://github.com/OpenIndiana/oi-userland/pull/4034). All 2 cores (+HT) are running at 50 %. MPEG-4 (xvid) is fine (1/10 files I have here locally). VLC 2 with FFMPEG 2 accelerate both, all core on 10-20 %.

**Comment by pyhalov** [comment]
@Mno-hime do you see something suspicious in vlc -vvv file_name output?
Can you reproduce issue just with ffmpeg / ffplay?

**Comment by pyhalov** [comment]
In vlc output we see two issues:
1) known one - hardware acceleration is disabled when number of threads != 1 (can be healed with setting  avcodec-threads to 1 in ~/.config/vlc/vlcrc)
2) ```Failed to open VDPAU backend ld.so.1: vlc: fatal: libvdpau_va_gl.so: open failed: No such file or directory``` - I suspect that Intel VA support libraries, tools and drivers were never ported to Solaris/illumos.

**Comment by Mno-hime** [comment]
On VLC2 `X11 video output (xcb)` is selected automatically:
```
core video output debug: Opening vout display wrapper
core vout display debug: looking for vout display module matching "any": 10 candidates
xcb vout display debug: connected to X11.0 server
xcb vout display debug:  vendor : Project OpenIndiana
xcb vout display debug:  version: 11907000
xcb vout display debug: using screen 0xea
vdpau_display vout display debug: device creation failure: error 1
xcb vout display debug: connected to X11.0 server
xcb vout display debug:  vendor : Project OpenIndiana
xcb vout display debug:  version: 11907000
xcb vout display debug: using screen 0xea
xcb_xv vout display debug: using XVideo extension v2.2
xcb_xv vout display debug: using adaptor Intel(R) Textured Video
xcb_xv vout display debug: using port 102
xcb_xv vout display debug: using image format 0x30323449
xcb_xv vout display debug: using X11 visual ID 0x21 (depth: 24)
xcb_xv vout display debug: using X11 window 0x04c00000
xcb_xv vout display debug: using X11 graphic context 0x04c00002
core vout display debug: VoutDisplayEvent 'fullscreen' 1
core vout display debug: VoutDisplayEvent 'resize' 600x341 fullscreen
core vout display debug: using vout display module "xcb_xv"
core vout display debug: A filter to adapt decoder to display is needed
core vout display debug: Filter 'Swscale' (bfdc18) appended to chain
xcb vout display debug: display is visible
xcb vout display debug: display is visible
core vout display debug: VoutDisplayEvent 'resize' 600x420 fullscreen
core vout display debug: VoutDisplayEvent 'resize' 1920x1080 fullscreen
core vout display debug: auto hiding mouse cursor
core vout display debug: Filter bfdc18 removed from chain
core vout display debug: removing module "xcb_xv"
```

With VLC3 `opengl` output is selected automatically:
```
main video output debug: Opening vout display wrapper
main vout display debug: looking for vout display module matching "any": 9 candidates
main vout display debug: VoutDisplayEvent 'resize' 1280x720
main vout display debug: VoutDisplayEvent 'resize' 1920x1006
main vout display debug: VoutDisplayEvent 'resize' 1920x1080
main vout display debug: using vout display module "gl"
main vout display debug: auto hiding mouse cursor
main vout display debug: removing module "gl"
```

Explicitly selecting `xcb` in VLC3 works as expected, all cores are around 30-40 %. So, OK from my POW. Setting `avcodec-threads` does not seems to make difference.

**Comment by pyhalov** [comment]
It seems to be the minimal change which shouldn't cause regressions (at least I hope so). 
Tested on Intel GM965.

**Comment by Mno-hime** [comment]
Added H.265 support, requires #5150 to be merged first.

**Comment by Mno-hime** [comment]
> It seems to be the minimal change which shouldn't cause regressions (at least I hope so).
> Tested on Intel GM965.

Works perfectly for me. I tested upgrade from VLC2 to VLC3 and VLC3 start without prior configuration - in both cases XCB is used and that works fine.

vlc-cache-gen is fixed for me as well.

**Review by Mno-hime — APPROVED** [review]
Works great with Intel Ivy Bridge GFX.

**Pull request merged**